benchmarks/kernels/tuning_utils.py

```diff
@@ -0,0 +1,128 @@
+from itertools import product
+
+
+## Utilize method from rocm/Triton tuning script
+def get_full_tuning_space():
+    configs = []
+
+    block_mn_range = [16, 32, 64, 128, 256]
+    block_k_range = [16, 32, 64, 128, 256]
+    # split_k_range = [1] #, 2, 4, 5, 6, 8, 10, 12, 16, 18, 24]
+    num_warps_range = [1, 2, 4, 8]
+    group_m_range = [1, 4, 8, 16, 32]
+    # For now we see better perf with num_stages=0 for all gemm configs we care
+    # But keep this explicit so that we do not forget we may need to set it to
+    # other values in the future
+    num_stage_range = [0]
```

Comment:
So this PR comes at a tricky time unfortunately.

See https://github.com/triton-lang/triton/pull/4845#pullrequestreview-2376204521

We are changing the sw pipelining such that it is more aligned with the nvidia side. After the above PR is merged (imminently), num_stages = 0 will actually be num_stages=2. If it is kept at 0, it will fail with an error as mentioned in the above link.

So we have two options

1) Hold this PR, if possible, until the Triton PR is submitted.

2) Submit this and then submit another once things break. As mentioned, the error message should be clear and should say what to do.

Reply:
@shajrawi FYI